benlk/republication-tracker-tool
# Add shortcode that outputs the republication button modal

## Changes

This pull request makes the following changes:

- Refactors the Republication Tracker Tool widget to create two new static functions:
    - `Republication_Tracker_Tool_Widget::button_output( $label = '' );` when echoed returns the button of the widget that opens the modal, with the optional parameter`$label` that changes the button text from the default
    - `Republication_Tracker_Tool_Widget::maybe_print_modal_content();` which, when echoed, if the modal content has not already been output on the page, does so.
- Adds a `[republication_modal_button label=""]` shortcode that outputs a button that will open the modal, with the optional parameter `label=""` to change the button text
- Reduces selector specificity for widget-provided styles on `button.republication-tracker-tool-button`
- Makes notes about all of that in the `docs/` directory
- The property `Republication_Tracker_Tool_Widget->has_instance` is now `Republication_Tracker_Tool->has_instance`.

## Why



For https://github.com/INN/republication-tracker-tool/issues/66 and https://github.com/PublicI/umbrella-public-integrity/issues/65

## Testing/Questions

Features that this PR affects:

- The Block Editor: adds a block
- The post editor: adds a shortcode
- The Widget output: changes some of how the widget is output upon the page
- The property `Republication_Tracker_Tool_Widget->has_instance` is now `Republication_Tracker_Tool->has_instance`, which may have effects on third-party code.


Questions that need to be answered before merging:

- [x] Does this work in Largo and outside of it?
     - I've tested it on Largo and on the publicintegrity.org theme
- [ ] Are you worried about the property `Republication_Tracker_Tool_Widget->has_instance` moving to `Republication_Tracker_Tool->has_instance`?

Steps to test this PR:

1. Add some shortcodes to a post content:
    - `[republication_modal_button label=""]`
    - `[republication_modal_button label="Cheese"]`
    - `[republication_modal_button label="Llanfair­pwllgwyngyll­gogery­chwyrn­drobwll­llan­tysilio­gogo­goch"]`
    - `[republication_modal_button label="The Republication Tracker Tool is designed to make sharing your content easier."]`
    - `[republication_modal_button]` 
2. Add some Republication Modal Button blocks to some posts.
3. Add some Republication Tracker Tool widgets to some widget areas on post pages.
4. Visit the post pages and make sure that nothing is horribly broken.

## Additional information

INN Member/Labs Client requesting: Center for Public Integrity

- [x] Contributor would like to be mentioned in the release notes as: The Center for Public Integrity
